packages/autoloader/src/AutoloadProcessor.php

```diff
@@ -0,0 +1,162 @@
+<?php // phpcs:ignore WordPress.Files.FileName
+/**
+ * Autoload Processor.
+ *
+ * @package automattic/jetpack-autoloader
+ */
+
+// phpcs:disable WordPress.Files.FileName.InvalidClassFileName
+// phpcs:disable WordPress.NamingConventions.ValidFunctionName.MethodNameInvalid
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.InterpolatedVariableNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.PropertyNotSnakeCase
+
+namespace Automattic\Jetpack\Autoloader;
+
+use Composer\Util\Filesystem;
+
+/**
+ * Class AutoloadProcessor.
+ */
+class AutoloadProcessor {
+
+	/**
+	 * A callable for scanning a directory for all of its classes.
+	 *
+	 * @var callable
+	 */
+	private $classmapScanner;
+
+	/**
+	 * A callable for transforming a path into one to be used in code.
+	 *
+	 * @var callable
+	 */
+	private $pathCodeTransformer;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param callable $classmapScanner A callable for scanning a directory for all of its classes.
+	 * @param callable $pathCodeTransformer A callable for transforming a path into one to be used in code.
+	 */
+	public function __construct( $classmapScanner, $pathCodeTransformer ) {
+		$this->classmapScanner     = $classmapScanner;
+		$this->pathCodeTransformer = $pathCodeTransformer;
+	}
+
+	/**
+	 * Processes the classmap autoloads into a relative path format including the version for each file.
+	 *
+	 * @param array $autoloads The autoloads we are processing.
+	 * @param bool  $scanPsrPackages Whether or not PSR packages should be converted to a classmap.
+	 *
+	 * @return array $processed
+	 */
+	public function processClassmap( $autoloads, $scanPsrPackages ) {
+		// We can't scan PSR packages if we don't actually have any.
+		if ( empty( $autoloads['psr-4'] ) ) {
+			$scanPsrPackages = false;
+		}
+
+		if ( empty( $autoloads['classmap'] ) && ! $scanPsrPackages ) {
+			return null;
+		}
+
+		$classBlacklist = null;
```

Comment:
What do you think of naming this something like `$excludeList`? We've been working on removing terms like 'blacklist', as described in [this issue](https://github.com/Automattic/jetpack/issues/16099).

Reply:
I've gone with `$excludedClasses` to keep what its excluding clear, thanks!